### ISSUE TITLE
m4v files not play

### ISSUE THREAD

**Issue by gpopesc** [mattermost/desktop#1492]
m4v video files (sent from whatsapp) are not played automatically in mattermost window 
I can download them locally and play or another option: copy external link and past in browser.

They are played automatically in android version of mattermost

Thank you

**Comment by amyblais** [comment]
@gpopesc What Mattermost desktop app version and Mattermost server version are you using? What OS and version is this on?

**Comment by gpopesc** [comment]
Mattermost windows version : 4.61
Windows 10 ver 18363 and windows 7
Mattermost server: 5:32

I installed matterbridge and I get a lot of video files. mp4 can be played, m4v it can't

**Comment by amyblais** [comment]
I can reproduce but not sure if it has always been like this. @Willyfrog Do you know if this is a bug or a "feature request"?

**Comment by Willyfrog** [comment]
it would be a feature request for web app, I believe it might depend on the browser you are in that it is enabled out of the box

**Comment by vegerot** [comment]
I'm trying to find the place in the source code where it checks whether to play the video or just prompt for download.  I see https://github.com/mattermost/mattermost-webapp/blob/b670eca198c5e5a97878bf85160d480ead20ae61/e2e/cypress/integration/files_and_attachments/helpers.js#L45 it "checks if the video element exists".

Could it be for some reason that running in the `<webview />` Mattermost cannot determine if the video element exists?

As a test, I opened Mattermost in the browser, previewed a video, copied the HTML of the modal, and pasted it into the desktop app, and it worked perfectly.  So there is nothing wrong with playing videos in Electron.  All we need to do is fix the detection bug.

**Comment by vegerot** [comment]
I have figured it out: 

tl;dr: Chromium does not support `.mov` files, but Safari does

Upon looking a bit more, it looks like whether it shows the video preview or not happens here https://github.com/mattermost/mattermost-webapp/blob/master/components/audio_video_preview.jsx#L84-L91 .  This is determined from here https://github.com/mattermost/mattermost-webapp/blob/master/components/audio_video_preview.jsx#L62-L67 .  

So: whether or not the video plays is determined by whether 

```js
document.createElement('video').canPlayType(fileInfo.mime_type);
```

is `maybe`/`probably` or `""`.  You can see the descriptions for these values on [HTMLMediaElement.canPlayType()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canPlayType).

I was using a `.mov` file from my Mac.  The file played correctly on Safari because 

```js
document.createElement('video').canPlayType('video/quicktime');
```
( see [container formats](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#quicktime) )
is truthy.

However, running this in Chrome is falsy.  So in both the Mattermost web app in Chrome and the Electron client, .mov files will not play.

In my earlier post, I accidentally introduced error into my experiment by comparing the Mattermost Desktop apps and the Safari web app; instead of Chrome.

@gpopesc make sure Chrome supports the format of video files you want to play.

**Comment by gpopesc** [comment]
@vegerot Yes, is exactly what I am doing: I copy the link as external link and paste in browser. It works in both chrome and firefox.
you can take a sample from[ here](https://filesamples.com/formats/m4v) and send it. In mattermost for windows asks for downloading or get external link. In mattermost for android it plays straight away.

**Comment by vegerot** [comment]
@gpopesc try seeing how mattermost interprets the file, and whether it thinks that the mime_type is playable

**Comment by gpopesc** [comment]
if I run in console log:
var obj = document.createElement('video');
console.log(obj.canPlayType('video/mp4'));
I get: VM146:2 maybe

if I run:
var obj = document.createElement('video');
console.log(obj.canPlayType('video/m4v'));
the answer is empty

if I run: 
document.createElement('video').canPlayType('video/quicktime');
""
document.createElement('video').canPlayType('video/mp4');
"maybe"
document.createElement('video').canPlayType('video/m4v');
""

if I run
document.createElement('video').canPlayType(fileInfo.mime_type);
VM231:1 Uncaught ReferenceError: fileInfo is not defined
    at <anonymous>:1:45
